esphome/components/openthread/__init__.py

```diff
@@ -82,7 +88,17 @@ def set_sdkconfig_options(config):
     add_idf_sdkconfig_option("CONFIG_OPENTHREAD_SRP_CLIENT_MAX_SERVICES", 5)
 
     # TODO: Add suport for sleepy end devices
-    add_idf_sdkconfig_option("CONFIG_OPENTHREAD_FTD", True)  # Full Thread Device
+    if dev_type := config.get(CONF_DEVICE_TYPE):
+        if dev_type == "MTD":
+            add_idf_sdkconfig_option(
+                "CONFIG_OPENTHREAD_MTD", True
+            )  # Minimum Thread Device
+        else:
+            add_idf_sdkconfig_option(
+                "CONFIG_OPENTHREAD_FTD", True
+            )  # Full Thread Device
+    else:
+        add_idf_sdkconfig_option("CONFIG_OPENTHREAD_FTD", True)  # Full Thread Device
```

Comment:
Doesnt need to be so complex :wink: 
```suggestion
    add_idf_sdkconfig_option(f"CONFIG_OPENTHREAD_{config.get(CONF_DEVICE_TYPE)}", True)
```